test/service/test_request_validation.py

```diff
@@ -433,18 +427,26 @@ def test_missing_uuid(self):
 
     def test_default_for_missing_params(self):
         path = '/test/mock/endpoints'
-        method_spec = {'parameters': [],
-                       'responses': {'200': {'description': 'OK'}}, 'tags': ['Index']}
-
-        for required, catalog in product((True, False), (self.catalog, None)):
-            param_spec = {'in': 'query', 'schema': {
-                'type': 'string',
-                'pattern': '^([a-z0-9]{1,64})$',
-                'enum': ['test']
-            }, 'required': required, 'name': f'required-{required}-default-{catalog}'}
-            if catalog is not None:
-                param_spec['schema']['default'] = catalog
-            method_spec['parameters'].append(param_spec)
+        method_spec = {
+            'parameters': [
+                {
+                    'in': 'query',
+                    'schema': {
+                        'type': 'string',
+                        'pattern': '^([a-z0-9]{1,64})$',
+                        'enum': [self.catalog],
+                        **({} if default is None else {'default': default})
+                    },
+                    'required': required,
+                    'name': f'required-{required}-default-{default}'
+                } for required, default in product((True, False), (self.catalog, None))
```

Comment:
This is purely suggestion, but this method is also slightly shorter.
```suggestion
                } for required in (True, False) for default in (self.catalog, None)
```